commit a3a2783237402f8379a5b950a63b7be93afb1494
Author: Dave Airlie <airlied@redhat.com>
Date:   Thu Mar 25 08:34:28 2021 +1000

    drisw: move zink down the list below the sw drivers.
    
    We don't ever want drisw path picking zink as the driver,
    we can revisit this when the penny wrapper work gets further
    along.
    
    This selection causes systems with nvidia/intel dual-gpus
    to try and pick the intel gpu for rendering in the nvidia
    context if there is no nvidia GL driver or accel doesn't work.
    
    This is a partial revert of the original commit.
    
    Fixes: 4a3b42a717ce ("drisw: Prefer hardware-layered sw-winsys drivers over pure sw")
    Acked-by: Jesse Natalie <jenatali@microsoft.com>
    Reviewed-by: Adam Jackson <ajax@redhat.com>
    Acked-By: Mike Blumenkrantz <michael.blumenkrantz@gmail.com>
    Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/9816>
    (cherry picked from commit 3e1698fe1baa22d000e29bd267b1c39b59910cca)

--- a/.pick_status.json
+++ b/.pick_status.json
@@ -1462,7 +1462,7 @@
         "description": "radv: Dedupe winsyses per device.",
         "nominated": true,
         "nomination_type": 1,
-        "resolution": 0,
+        "resolution": 1,
         "master_sha": null,
         "because_sha": "7bef19201822ab2aebfd244142ff3a23535019a7"
     },
--- a/src/gallium/auxiliary/target-helpers/inline_sw_helper.h
+++ b/src/gallium/auxiliary/target-helpers/inline_sw_helper.h
@@ -81,9 +81,6 @@ sw_screen_create(struct sw_winsys *winsy
    UNUSED bool only_sw = env_var_as_boolean("LIBGL_ALWAYS_SOFTWARE", false);
    const char *drivers[] = {
       debug_get_option("GALLIUM_DRIVER", ""),
-#if defined(GALLIUM_ZINK)
-      only_sw ? "" : "zink",
-#endif
 #if defined(GALLIUM_D3D12)
       only_sw ? "" : "d3d12",
 #endif
@@ -96,6 +93,9 @@ sw_screen_create(struct sw_winsys *winsy
 #if defined(GALLIUM_SWR)
       "swr",
 #endif
+#if defined(GALLIUM_ZINK)
+      only_sw ? "" : "zink",
+#endif
    };
 
    for (unsigned i = 0; i < ARRAY_SIZE(drivers); i++) {
--- a/src/gallium/auxiliary/target-helpers/sw_helper.h
+++ b/src/gallium/auxiliary/target-helpers/sw_helper.h
@@ -86,9 +86,6 @@ sw_screen_create(struct sw_winsys *winsy
    UNUSED bool only_sw = env_var_as_boolean("LIBGL_ALWAYS_SOFTWARE", false);
    const char *drivers[] = {
       debug_get_option("GALLIUM_DRIVER", ""),
-#if defined(GALLIUM_ZINK)
-      only_sw ? "" : "zink",
-#endif
 #if defined(GALLIUM_D3D12)
       only_sw ? "" : "d3d12",
 #endif
@@ -101,6 +98,9 @@ sw_screen_create(struct sw_winsys *winsy
 #if defined(GALLIUM_SWR)
       "swr",
 #endif
+#if defined(GALLIUM_ZINK)
+      only_sw ? "" : "zink",
+#endif
    };
 
    for (unsigned i = 0; i < ARRAY_SIZE(drivers); i++) {
